plugins/fetcher/prometheus/fetcher.go

```diff
@@ -0,0 +1,146 @@
+// Licensed to Apache Software Foundation (ASF) under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Apache Software Foundation (ASF) licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package prometheus
+
+import (
+	"context"
+	"time"
+
+	"github.com/apache/skywalking-satellite/internal/pkg/config"
+	"github.com/apache/skywalking-satellite/internal/pkg/log"
+	"github.com/apache/skywalking-satellite/internal/satellite/event"
+
+	promConfig "github.com/prometheus/prometheus/config"
+	"github.com/prometheus/prometheus/discovery"
+	"github.com/prometheus/prometheus/scrape"
+	yaml "gopkg.in/yaml.v3"
+	v1 "skywalking.apache.org/repo/goapi/satellite/data/v1"
+)
+
+const (
+	Name      = "prometheus-metrics-fetcher"
+	eventName = "prometheus-metrics-event"
+)
+
+type scrapeConfig struct {
+	JobName        string                   `yaml:"job_name" mapstructure:"job_name"`
+	ScrapeInterval time.Duration            `yaml:"scrape_interval,omitempty" mapstructure:"scrape_interval,omitempty"`
+	StaticConfigs  []map[string]interface{} `yaml:"static_configs" mapstructure:"static_configs"`
+	MetricsPath    string                   `yaml:"metrics_path,omitempty" mapstructure:"metrics_path,omitempty"`
+}
+
+// Fetcher is the struct for Prometheus fetcher
+type Fetcher struct {
+	config.CommonFields
+	// config is the top level configuratScrapeConfigsMapion of prometheus
+	ScrapeConfigsMap []*scrapeConfig `mapstructure:"scrape_configs" yaml:"scrape_configs"`
+
+	ScrapeConfigs []*promConfig.ScrapeConfig
+	// events
+	OutputEvents event.BatchEvents
+	// outputChannel
+	OutputChannel chan *v1.SniffData
+
+	cancelFunc context.CancelFunc
+}
+
+func (f *Fetcher) Name() string {
+	return Name
+}
+
+func (f *Fetcher) Description() string {
+	return "This is a fetcher for Skywalking prometheus metrics format, " +
+		"which will translate Prometheus metrics to Skywalking meter system."
+}
+
+func (f *Fetcher) DefaultConfig() string {
+	return `
+## some config here
+scrape_configs:
+ - job_name: 'prometheus'
+   metrics_path: '/metrics'
+   scrape_interval: 10s
+   static_configs:
+   - targets: ['127.0.0.1:2020']
+`
+}
+
+func (f *Fetcher) Prepare() {}
+
+func (f *Fetcher) Fetch() event.BatchEvents {
```

Comment:
I can't see when this method will be called? Do I miss anything?

Reply:
@gxthrj Should we remove this legacy API? @mrproliu This method was called in the core level timer, which has been removed.

Reply:
There is an error in fetcher_gather.go, the fetch() call is missing. I will add it.

Reply:
Done.

Reply:
> @gxthrj Should we remove this legacy API? @mrproliu This method was called in the core level timer, which has been removed.

the Fech() is still need when booting.

Reply:
I have modified the Fetcher interface
1. remove the response, which is not needed any more.
2. add `ctx` for Fetch(ctx) , which used in conjunction with shutdown(ctx).

cc @EvanLjp